src/lib/core/strftime.c

```diff
@@ -0,0 +1,685 @@
+					int wday;
+					int w;
+
+					year = t->tm_year;
+					base = TM_YEAR_BASE;
+					yday = t->tm_yday;
+					wday = t->tm_wday;
+					for (;;) {
+						int len;
+						int bot;
+						int top;
+
+						len = isleap_sum(year, base)
+						    ? DAYSPERLYEAR
+						    : DAYSPERNYEAR;
+						/*
+						 ** What yday (-3 ... 3) does
+						 ** the ISO year begin on?
+						 */
+						bot = ((yday + 11 - wday) %
+						       DAYSPERWEEK) - 3;
+						/*
+						 ** What yday does the NEXT
+						 ** ISO year begin on?
+						 */
+						top = bot - (len % DAYSPERWEEK);
+						if (top < -3)
+							top += DAYSPERWEEK;
+						top += len;
+						if (yday >= top) {
+							++base;
+							w = 1;
+							break;
+						}
+						if (yday >= bot) {
+							w = 1 + ((yday - bot) /
+								 DAYSPERWEEK);
+							break;
+						}
+						--base;
+						yday += isleap_sum(year, base)
+						    ? DAYSPERLYEAR
+						    : DAYSPERNYEAR;
+					}
+#ifdef XPG4_1994_04_09
+					if ((w == 52 &&
+					     t->tm_mon == TM_JANUARY) ||
+					    (w == 1 &&
+					     t->tm_mon == TM_DECEMBER))
+						w = 53;
+#endif				/* defined XPG4_1994_04_09 */
+					if (*format == 'V')
+						pt = _conv(w, "%02d",
+							   pt, ptlim);
+					else if (*format == 'g') {
+						*warnp = IN_ALL;
+						pt = _yconv(year, base,
+							    false, true,
+							    pt, ptlim);
+					} else
+						pt = _yconv(year, base,
+							    true, true,
+							    pt, ptlim);
+				}
+				continue;
+			case 'v':
+				/*
+				 ** From Arnold Robbins' strftime version 3.0:
+				 ** "date as dd-bbb-YYYY"
+				 ** (ado, 1993-05-24)
+				 */
+				pt = _fmt("%e-%b-%Y", t, pt, ptlim, warnp);
+				continue;
+			case 'W':
+				pt = _conv((t->tm_yday + DAYSPERWEEK -
+					    (t->tm_wday ? (t->tm_wday - 1)
+					     : (DAYSPERWEEK -
+						1))) /
+					   DAYSPERWEEK, "%02d", pt, ptlim);
+				continue;
+			case 'w':
+				pt = _conv(t->tm_wday, "%d", pt, ptlim);
+				continue;
+			case 'X':
+				pt = _fmt(Locale->X_fmt, t, pt, ptlim, warnp);
+				continue;
+			case 'x':
+				{
+					enum warn warn2 = IN_SOME;
+
+					pt = _fmt(Locale->x_fmt, t, pt, ptlim,
+						  &warn2);
+					if (warn2 == IN_ALL)
+						warn2 = IN_THIS;
+					if (warn2 > *warnp)
+						*warnp = warn2;
+				}
+				continue;
+			case 'y':
+				*warnp = IN_ALL;
+				pt = _yconv(t->tm_year, TM_YEAR_BASE,
+					    false, true, pt, ptlim);
+				continue;
+			case 'Y':
+				pt = _yconv(t->tm_year, TM_YEAR_BASE,
+					    true, true, pt, ptlim);
+				continue;
+			case 'Z':
+				if (t->tm_isdst >= 0)
+					pt = _add(tzname[t->tm_isdst != 0],
+						  pt, ptlim);
+				/*
+				 ** C99 and later say that %Z must be
+				 ** replaced by the empty string if the
+				 ** time zone abbreviation is not
+				 ** determinable.
+				 */
+				continue;
+			case 'z':
+				{
+					long diff;
+					char const *sign;
+					bool negative;
+
+					diff = t->tm_gmtoff;
+					negative = diff < 0;
+					if (diff == 0) {
+						negative = t->tm_isdst < 0;
```

Comment:
I've removed this code ,to make it behave consistently across platforms (some targets had '-' in `tzname[]`, but others not). I believe that our code should not rely on externally controlled `tzname`, and we will control timezone information fully internally, using build artifacts. 

```diff
@@ -594,12 +594,6 @@ static char *_fmt(const char *format, const struct dt_tm *t, char *pt,
 
 					diff = t->tm_gmtoff;
 					negative = diff < 0;
-					if (diff == 0) {
-						negative = t->tm_isdst < 0;
-						if (tzname[t->tm_isdst != 0][0]
-						    == '-')
-							negative = true;
-					}
 					if (negative) {
 						sign = "-";
 						diff = -diff;
```